src/services/ldap-config/service.js

```diff
@@ -0,0 +1,288 @@
+const mongoose = require('mongoose');
+const reqlib = require('app-root-path').require;
+
+const errorHandlers = require('./errors');
+
+const { Forbidden } = reqlib('src/errors');
+
+class LdapConfigService {
+	setup(app) {
+		this.app = app;
+		this.systemId = undefined;
+	}
+
+	/**
+	 * `GET /ldap-config/:systemId`
+	 * Returns the ldapConfig of a given system in the same format as accepted
+	 * by the other methods of this service.
+	 * If the given system is not a LDAP system (i.e. does not have the ldapConfig
+	 * attribute) or is not of the type `general` (i.e. a self-service config), the
+	 * access is denied.
+	 * To access the system data, the authentication information are proxied to the
+	 * System service, so that access is only granted to administrators and only to
+	 * systems of the own school.
+	 * @param {ObjectId} systemId ID of an existing system with LDAP config
+	 * @param {Object} params Feathers request params
+	 * @returns {Object} ldapConfig attribute of the given system
+	 */
+	async get(systemId, params) {
+		// Proxying the request params here to have the system service hooks check
+		// the permissions, filtering sensitive data, etc.
+		const system = await this.app.service('systems').get(systemId, params);
+		if (!system.ldapConfig || system.ldapConfig.provider !== 'general') {
+			throw new Forbidden('The requested system is not configurable with this endpoint');
+		}
+		return system.ldapConfig;
+	}
+
+	/**
+	 * `POST /ldap-config`
+	 * Creates a new system with the given LDAP config and attaches it to the
+	 * caller's school, if the config is validated successfully. The validation
+	 * result is returned.
+	 * If `validateOnly === true`, changes are not persisted to the systems and
+	 * school collection. Otherwise, a new system is created.
+	 * After creating the system, it is added as a login system to the caller's
+	 * school and (if `activate !== false`) the school's ldapSchoolIdentifier
+	 * attribute is set to the config's base path.
+	 * @param {Object} config an LDAP config (given as JSON in request body)
+	 * @param {Object} params Feathers request params
+	 * @returns {Object} validation result object
+	 * `{ ok: Boolean, errors: [], users: {}, classes: {} }`
+	 */
+	async create(config, params) {
+		return this.verifyAndSaveLdapConfig(config, this.getOptions(params));
```

Comment:
Please see getOptions comment.

Reply:
If you would use concrete params, then it would be not necessary to explain so much in the docs :)

Reply:
not sure if this aplies here. The method signature is given by Feathers